File: Python/tempAnalyse.py

temps = [0]
sum = 0
u30 = 0
u4 = 0
middletemps = 0

print("Please enter the temperatures one by one!")

for x in range(7):
    value = input("Temperature: ")
    temps.append(value)

temps.pop(0)

for x in temps:
    sum = sum + int(x)

average = sum / len(temps)

for x in temps:
    if int(x) > 30:
        u30 = u30 + 1
    if int(x) < 4:
        u4 = u4 + 1
    else:
        middletemps = middletemps + 1
        

print("Here is the collected data:")
print("The temperatures in the past week were: " + str(temps))
print("The average temperature was: " + str(average))
print("The number of temperatures above 30 was: " + str(u30))
print("The number of temperatures under 4 was: " + str(u4))
print("The number of temperatures between 4 and 30 was: " + str(middletemps))

File: Java/tempAnalyse.java
import java.util.*;

public class tempAnalyse {
    public static void main(String[] args) {
        try (Scanner temp = new Scanner(System.in)) {
            int[] temps = new int[7];
            int min=temps[0];
            int max=temps[0];
            int u30=0;
            int u4=0;

            for (int i = 0; i < temps.length; i++) {
                System.out.print("Put in the " + (i+1) + " value: ");
                int value = temp.nextInt();
                temps[i] = value;
            }

            int sum = 0;
            for (int i = 0; i < temps.length; i++) {
                sum += temps[i];
            }
            int average = sum / temps.length;
            
            for(int j=0; j<temps.length; j++){

                if (temps[j]>max) {
                    max = temps[j];
                }
                if (temps[j]<min) {
                    min=temps[j];
                }
                if (temps[j]>30) {
                    u30++;                    
                }
                if (temps[j]<4) {
                    u4++;                    
                }
            }
            
            System.out.println("Here is the collected data:");
            System.out.println("The temperatures in the last seven days were " + Arrays.toString(temps) + ".");
            System.out.println("Average temperatur in the last 7 days was " + average + "°C.");
            System.out.println("It was hotter than 30°C on " + u30 + " days.");
            System.out.println("It was colder than 4°C on " + u4 + " days");
        }
    }
}
